core/embed/firmware/bl_check.c

```diff
@@ -174,16 +208,31 @@ void check_and_replace_bootloader(void) {
 
   ensure(flash_area_erase(&BOOTLOADER_AREA, NULL), NULL);
   ensure(flash_unlock_write(), NULL);
-  for (int i = 0; i < len / sizeof(uint32_t); i++) {
-    ensure(
-        flash_area_write_word(&BOOTLOADER_AREA, i * sizeof(uint32_t), data[i]),
-        NULL);
-  }
-  for (int i = len / sizeof(uint32_t); i < 128 * 1024 / sizeof(uint32_t); i++) {
-    ensure(flash_area_write_word(&BOOTLOADER_AREA, i * sizeof(uint32_t),
-                                 0x00000000),
-           NULL);
+
+  uint32_t offset = 0;
+
+  do {
+    uint32_t *p = decomp_out;
+    while ((uint32_t)p < (((uint32_t)decomp.dest) & ~3)) {
```

Comment:
`(((uint32_t)decomp.dest) & ~3)`

please spin this out into a local variable with a descriptive name

Reply:
fixed https://github.com/trezor/trezor-firmware/pull/3304/commits/4ddf5b306c2cedbbacdc4aad9c8bfe86f9432184